### PR TITLE
Update CHANGELOG

### PR DESCRIPTION
**Pull request by beseven**
## Description

Prepare release and add details to CHANGELOG

## Related Issue

Trying to push out the next release:  https://github.com/nhsuk/gp-finder/releases/tag/untagged-4f21b9481f1779cd41a5 

## Motivation and Context

Non-functional changes: Upgrades, ADR context and performance threshold

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Provide a general summary of your changes in the Title
 
 ## Checklist
 <!-- Ensure each of the points below have been considered and completed where applicable -->
-- [ ] An [ADR](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions) has been created and any deprecated/superceeded ADR's have had their status updated.
+- [ ] An [ADR](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions) has been created and any deprecated/superseded ADR's have had their status updated.
 - [ ] New and/or updated tests
 - [ ] Version of the cache is increased
+- [ ] Updated CHANGELOG (add features per PR basis, add date when TMR is being prepped)
```
Comment (st3v3nhunt) [inline .github/PULL_REQUEST_TEMPLATE.md]: Rather than `TMR`, `release` would be more generic

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
-0.17.0 / TBC
+0.16.2 / 2017-10-3
 ===================
-- Update all available npm dependencies
+- Update ADR guidance/superseded mongo ADR
+- Update all stable npm dependencies
+- Use threshold file for performance testing for baseline
+- Update CHANGELOG
```
Comment (st3v3nhunt) [inline CHANGELOG.md]: The changing of the `CHANGELOG` is implicit, I'd remove it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
-0.17.0 / TBC
+0.16.2 / 2017-10-3
 ===================
-- Update all available npm dependencies
+- Update ADR guidance/superseded mongo ADR
+- Update all stable npm dependencies
```
Comment (st3v3nhunt) [inline CHANGELOG.md]: The inclusion of `stable` makes me wonder what the unstable dependencies were and why they weren't updated.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
-0.17.0 / TBC
+0.16.2 / 2017-10-3
 ===================
-- Update all available npm dependencies
+- Update ADR guidance/superseded mongo ADR
```
Comment (st3v3nhunt) [inline CHANGELOG.md]: I wouldn't bother including a note about this change. No functionality has changed, just that documentation. As a third party I'm only interested in changes of interest and that piece of documentation isn't interesting as the functional change has been already.

File: CHANGELOG.md (hunk 3 above)
Comment (beseven) [reply CHANGELOG.md]: There was a discussion about updating node to `8.6.0` and why that wasn't done, but I guess that is distinct from dependencies?

Comment (st3v3nhunt) [reply CHANGELOG.md]: Yeah, it isn't an npm dependency and the result was no change to the code base/application so there'd be no need to mention it in the release notes.

**Pull request merged**